include/irr/asset/filters/CSummedAreaTableImageFilter.h

```diff
@@ -0,0 +1,398 @@
+				return false;
+
+			return true;
+		}
+
+		static inline bool execute(state_type* state)
+		{
+			if (!validate(state))
+			{
+				state->imageTotalCacheScratch = nullptr;
+				state->imageTotalCacheOutput = nullptr;
+				return false;
+			}
+
+			auto checkFormat = state->inImage->getCreationParameters().format;
+			if (isIntegerFormat(checkFormat))
+				return executeInterprated(state, reinterpret_cast<uint64_t*>(state->scratchMemory));
+			else
+				return executeInterprated(state, reinterpret_cast<double*>(state->scratchMemory));
+		}	
+
+	private:
+
+		template<typename decodeType> //!< double or uint64_t
+		static inline bool executeInterprated(state_type* state, decodeType* scratchMemory)
+		{
+			const asset::E_FORMAT inFormat = state->inImage->getCreationParameters().format;
+			const asset::E_FORMAT outFormat = state->outImage->getCreationParameters().format;
+			const auto inTexelByteSize = asset::getTexelOrBlockBytesize(inFormat);
+			const auto outTexelByteSize = asset::getTexelOrBlockBytesize(outFormat);
+			const auto currentChannelCount = asset::getFormatChannelCount(inFormat);
+			const auto arrayLayers = state->inImage->getCreationParameters().arrayLayers;
+			static constexpr auto maxChannels = 4u;
+
+			uint8_t* const totalImageCacheScratch = state->imageTotalCacheScratch = state->scratchMemory + state->cachesOffset;
+			uint8_t* const totalImageCacheOutput = state->imageTotalCacheOutput = state->imageTotalCacheScratch + state->imageTotalScratchCacheByteSize;
+
+			#ifdef _IRR_DEBUG
+			memset(scratchMemory, 0, state->scratchMemoryByteSize);
+			#endif // _IRR_DEBUG
+
+			const auto&& [copyInBaseLayer, copyOutBaseLayer, copyLayerCount] = std::make_tuple(state->inBaseLayer, state->outBaseLayer, state->layerCount);
+
+			auto resetState = [&]()
+			{
+				state->inBaseLayer = copyInBaseLayer;
+				state->outBaseLayer = copyOutBaseLayer;
+				state->layerCount = copyLayerCount;
+			};
+
+			core::vector3du32_SIMD localCoord;
+			for (auto& w = localCoord[3] = 0u; w < copyLayerCount - copyInBaseLayer; ++w) 
+			{
+				std::array<decodeType, maxChannels> minDecodeValues = {};
+				std::array<decodeType, maxChannels> maxDecodeValues = {};
+
+				state->inBaseLayer += w;
+				state->outBaseLayer += w;
+				state->layerCount = 1u;
+
+				auto perOutputRegionDecode = [&](const CommonExecuteData& commonExecuteData, CBasicImageFilterCommon::clip_region_functor_t& clip) -> bool
+				{
+					/*
+						Since regions may specify areas in a layer on a certain mipmap - it is desired to
+						decode an entire image itereting through all the regions overlapping as first
+					*/
+
+					const auto blockDims = asset::getBlockDimensions(commonExecuteData.inFormat);
+					const auto strides = commonExecuteData.oit->getByteStrides(TexelBlockInfo(commonExecuteData.inFormat)); // ITS WRONG
```

Comment:
ofc it's wrong, you're computing strides of output region with input image format info

Reply:
corrected in following commit